src/app/content/components/TableOfContents/keyboardSupport.hook.tsx

```diff
@@ -0,0 +1,171 @@
+import { HTMLElement, Element } from '@openstax/types/lib.dom';
+import { LinkedArchiveTreeNode } from '../../types';
+import React from 'react';
+
+export interface KeyboardSupportProps {
+  event: React.KeyboardEvent<HTMLAnchorElement>;
+  item: LinkedArchiveTreeNode;
+  isOpen?: boolean;
+  onSelect: () => void;
+}
+
+const getTreeItems = () => {
+  const treeItems = Array.from(document?.querySelectorAll('a[role="treeitem"]') as ArrayLike<HTMLElement>);
```

Comment:
use assertDocument() so that it throws if the logic is accidentally included in the nodejs render. you can pass the type to the `querySelectorAll` generic to avoid the cast